Mesh_3/include/CGAL/Polygon_mesh_processing/surface_Delaunay_remeshing.h

```diff
@@ -0,0 +1,304 @@
+*          (resp. a uniform) upper bound for the lengths of curve edges.
+*          This parameter has to be set to a positive value when 1-dimensional features protection is used
+*          (when `protect_constraints` is `true`).}
+*     \cgalParamType{A number type `FT` model of the concept `Field`, or a model of the concept `MeshDomainField_3`}
+*     \cgalParamDefault{`(std::numeric_limits<FT>::%max)()`, with
+*          `FT` a number type, either deduced from the `geom_traits` \ref bgl_namedparameters
+*          "Named Parameters" if provided, or from the geometric traits class deduced from the
+*          point property map of `TriangleMesh`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{mesh_facet_size}
+*     \cgalParamDescription{A scalar field (resp. a constant) describing a space-varying
+*          (resp. a uniform) upper bound for the radii of the surface Delaunay balls.}
+*     \cgalParamType{A number type `FT` model of the concept `Field`, or a model of the concept
+*          `MeshDomainField_3`}
+*     \cgalParamDefault{`0.`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{mesh_facet_angle}
+*     \cgalParamDescription{A lower bound for the angles (in degrees) of the surface mesh facets.}
+*     \cgalParamType{A number type `FT` model of the concept `Field`}
+*     \cgalParamDefault{`0.`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{mesh_facet_distance}
+*     \cgalParamDescription{A scalar field (resp. a constant) describing a space-varying
+*          (resp. a uniform) upper bound for the distance between the facet circumcenter
+*          and the center of its surface Delaunay ball.}
+*     \cgalParamType{A number type `FT` model of the concept `Field`, or a model of the concept
+*          `MeshDomainField_3`}
+*     \cgalParamDefault{`0.`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{mesh_facet_topology}
+*     \cgalParamDescription{The set of topological constraints which have to be verified
+*          by each surface facet. The default value is `CGAL::FACET_VERTICES_ON_SURFACE`.
+*          See `CGAL::Mesh_facet_topology` manual page for a description of the possible values.}
+*     \cgalParamType{\link PkgMesh3Enum `CGAL::Mesh_facet_topology` \endlink}
+*     \cgalParamDefault{`CGAL::FACET_VERTICES_ON_SURFACE`}
+*   \cgalParamNEnd
+* \cgalNamedParamsEnd
+*
+* \pre `tmesh` must be free of self-intersections.
+*
+* @note Only one of the named parameters defining constrained edges is taken into account
+* for meshing, in the following priority order : `edge_is_constrained_map`,
+* `polyline_constraints`, and `features_angle_bound`. The selected edges are protected only
+* if `protect_constraints` is set to `true`.
+*
+*/
+template<typename TriangleMesh
+       , typename TriangleMeshOut = TriangleMesh
+       , typename NamedParameters = parameters::Default_named_parameters>
+TriangleMeshOut surface_Delaunay_remeshing(const TriangleMesh& tmesh
```

Comment:
IMHO, the function should modify the input mesh directly (calling `clear()` when you're done with the remeshing). An alternative is to take the output as parameter but I don't really see any gain.

Reply:
IIRC taking the output as parameter was the initial version, and a discussion with @lrineau led to that version.
The benefit of modifying the input mesh is to use less space, is there another one?

Reply:
Sorry now I remember our discussion. So what you have is correct because you document that the  default VPM of out must have the same value type as the input and this works. Less flexible than having two meshes as input parameters but it works.

Reply:
Well you're missing that the value types must be compatible.